src/index.ts

```diff
@@ -17,19 +17,35 @@ class ServerlessFargatePlugin {
 
     private compile(): void {
         const service: any = this.serverless.service;
-        const options: IPluginOptions = service.custom.fargate;
+        const options: IClusterOptions[] = service.custom.fargate;
         const stage: string = service.provider ? service.provider.stage : service.stage;
-
-        // we could want more than one cluster in the future potentially per vpc
-        const vpc: VPC = new VPC(stage, options.vpc);
-        const cluster: Cluster = new Cluster(stage, options, vpc);
-
-        // merge all our stuff into resources
-        Object.assign(
-            this.serverless.service.provider.compiledCloudFormationTemplate.Resources,
-            vpc.generate(),
-            cluster.generate()
-        );
+        
+        //No cluster section specified, don't process
+        if (!options || !options.length) return;
+
+        //For each cluster
+        for (let clusterOption of options) {
+            if (clusterOption && clusterOption.vpc) { //sanity check for empty objects
+                const index = options.indexOf(clusterOption);
```

Comment:
if we need the index, we should just use basic for loop instead of for/of

Reply:
Unused index, removing it.